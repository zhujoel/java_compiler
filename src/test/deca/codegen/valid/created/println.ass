; Test pour savoir si la pile est pleine
	TSTO #8
	BOV pile_pleine
	ADDSP #6
; start main program
; Class Declaration
; Code de la table des méthodes de Object
	LOAD #null, R0
	STORE R0, 1(GB)
	LOAD code.Object.equals, R0
	STORE R0, 2(GB)
; Main Function
; Declaration des variables
	LOAD #0, R0
	STORE R0, 3(GB)
	LOAD #0x0.0p0, R0
	STORE R0, 4(GB)
	LOAD #0, R0
	STORE R0, 5(GB)
; Beginning of main instructions:
	WSTR "Hello"
	WNL
	WSTR "5"
	WNL
	LOAD #4, R2
	STORE R2, 3(GB)
	LOAD 3(GB), R1
	WINT
	WNL
	WSTR "2.0"
	WNL
	LOAD #0x1.333334p0, R2
	STORE R2, 4(GB)
	LOAD 4(GB), R1
	WFLOAT
	WNL
	HALT
code.Object.equals:
pile_pleine:
	WSTR "Erreur : Pile pleine."
	WNL
	ERROR
dereferencement_null:
	WSTR "Erreur : dereferencement null."
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : Tas plein."
	WNL
	ERROR
debordement:
	WSTR "Erreur : debordement durant une operation arithmétique"
	WNL
	ERROR
io_error:
	WSTR "Erreur : input ou output"
	WNL
	ERROR
; end main program
