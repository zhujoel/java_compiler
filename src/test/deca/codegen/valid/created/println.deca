{
    int a;
    float b;
    bool c;
    println("Hello");
    println(5);
    a = 4;
    println(a);
    println(2.0);
    b = 1.2;
    println(b);
}
