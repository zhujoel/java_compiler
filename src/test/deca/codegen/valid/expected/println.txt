Hello
5
4
2.0
1.20000e+00
